WireDomain/Sources/WireDomain/Notifications/GenerateNotificationService.swift

```diff
@@ -0,0 +1,170 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CallKit
+import UserNotifications
+import WireAPI
+import WireDataModel
+import WireLogging
+
+// sourcery: AutoMockable
+protocol GenerateNotificationServiceProtocol {
+    func process() async
+}
+
+struct GenerateNotificationService: GenerateNotificationServiceProtocol {
+
+    private let eventsStream: AsyncStream<[UpdateEvent]>
+    private let contentHandler: (UNNotificationContent) -> Void
+    private let accountManager: AccountManager
+    private let selectedAccount: Account
+    private let accountID: UUID
+    private let userDefaults: UserDefaults
+    private let userLocalStore: any UserLocalStoreProtocol
+    private let conversationLocalStore: any ConversationLocalStoreProtocol
+    private let messageLocalStore: any MessageLocalStoreProtocol
+
+    init(
+        eventsStream: AsyncStream<[UpdateEvent]>,
+        contentHandler: @escaping (UNNotificationContent) -> Void,
+        accountManager: AccountManager,
+        selectedAccount: Account,
+        accountID: UUID,
+        userDefaults: UserDefaults,
+        userLocalStore: any UserLocalStoreProtocol,
+        conversationLocalStore: any ConversationLocalStoreProtocol,
+        messageLocalStore: any MessageLocalStoreProtocol
+    ) {
+        self.eventsStream = eventsStream
+        self.contentHandler = contentHandler
+        self.accountManager = accountManager
+        self.selectedAccount = selectedAccount
+        self.accountID = accountID
+        self.userDefaults = userDefaults
+        self.conversationLocalStore = conversationLocalStore
+        self.userLocalStore = userLocalStore
+        self.messageLocalStore = messageLocalStore
+    }
+
+    /// Processes the events stream.
+    func process() async {
+        for await events in eventsStream {
+            await generateNotifications(for: events)
+        }
+    }
+
+    private func generateNotifications(
+        for events: [UpdateEvent]
+    ) async {
+        guard !events.isEmpty else {
+            return contentHandler(.emptyNotification)
```

Comment:
issue: suppose the stream is something like: `[event1], [], [event2]`. Then it would mean when we process the second element, we consume the event handler, even though we processed already one event successfully, and still have one more to go. Perhaps here we should simply skip the empty array and move to the next.